frame/balances/src/types.rs

```diff
@@ -0,0 +1,157 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types used in the pallet.
+
+use crate::{Config, CreditOf, Event, Pallet};
+use codec::{Decode, Encode, MaxEncodedLen};
+use core::ops::BitOr;
+use frame_support::{
+	traits::{Imbalance, LockIdentifier, OnUnbalanced, WithdrawReasons},
+	RuntimeDebug,
+};
+use scale_info::TypeInfo;
+use sp_runtime::Saturating;
+
+/// Simplified reasons for withdrawing balance.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, MaxEncodedLen, TypeInfo)]
+pub enum Reasons {
+	/// Paying system transaction fees.
+	Fee = 0,
+	/// Any reason other than paying system transaction fees.
+	Misc = 1,
```

Comment:
Why are the indices manually set differently?

Reply:
Legacy code https://github.com/paritytech/substrate/pull/12951/files/1d47200f500f3799dc940e9ff15b7cf1bdba90ab#diff-7f35e750129ddf17862e61be2ce506a120ac7d4de151633d2c026525130bba7dL629-L638. It hasn't changed in years - just repotted in this PR. Expect the indices there were just a measure of explicitness. It'll be removed along with `Currency` trait once we move everything to `fungible`.